railties/lib/rails/commands/server/server_command.rb

```diff
@@ -127,6 +127,14 @@ class ServerCommand < Base # :nodoc:
       class_option "dev-caching", aliases: "-C", type: :boolean, default: nil,
         desc: "Specifies whether to perform caching in development."
 
+      class << self
+        def restart_server
+          FileUtils.mkdir_p("tmp")
+          FileUtils.touch("tmp/run_restart.txt")
```

Comment:
It’d be clearer to include a flag that indicates a restart in the `restart_cmd`, then use that flag to determine whether to remove the PID file. Then we can keep `FileUtils.touch "tmp/restart.txt"` in `Rails::DevCaching.enable_by_file`.